cloud/deploy/deploy.go

```diff
@@ -879,8 +951,18 @@ func DeployClientImage(deployInput InputClientDeploy) error { //nolint:gocritic
 			fmt.Println("Building client image for host platform")
 		}
 
+		// Prepare build context with client dependency files
+		buildContext, err := prepareClientBuildContext(deployInput.Path)
+		if buildContext != nil {
+			defer buildContext.CleanupFunc()
+		}
```

Comment:
```suggestion
		if buildContext != nil && buildContext.CleanFunc != nil {
			defer buildContext.CleanupFunc()
		}
```
nit: double safety from panic :P